### PR TITLE
examples: Add ui_progress_updates

### PR DESCRIPTION
**Pull request by bkeryan**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Add an example demonstrating how to update the UI during the measurement.

This is an updated version of the example proposed in https://github.com/ni/measurementlink-python/pull/262 .

### Why should this Pull Request be merged?

Demonstrate the incremental update feature.

### What testing has been done?

Manually tested with latest pre-release InstrumentStudio and MeasurementLink UI Editor.

### CODE REVIEW

**Comment by bkeryan** [comment]
Screenshot:
image

File: examples/README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ This directory contains the following example projects:
 - `niscope_acquire_waveform`: Acquires a waveform using an NI oscilloscope.
 - `niswitch_control_relays`: Controls relays using an NI relay driver (e.g. PXI-2567).
 - `nivisa_dmm_measurement`: Performs a DMM measurement using NI-VISA and an NI Instrument Simulator v2.0.
+- `ui_progress_updates`: Generates random numbers for several seconds, updating the UI to show progress.
```
Comment (dixonjoel) [inline examples/README.md:14]: 'For a configurable number of seconds'?

**Review by pbirkhol-ni — APPROVED** [review]

**Review by dixonjoel — APPROVED** [review]

File: examples/ui_progress_updates/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+[tool.poetry.dependencies]
+python = "^3.8"
+ni-measurementlink-service = {version = "^1.1.0a0", allow-prereleases = true}
+click = ">=7.1.2"
+
+[tool.poetry.group.dev.dependencies]
+mypy = ">=1.0"
+# Uncomment to use prerelease dependencies (requires poetry>=1.2).
+# TODO: Comment this back out once 1.1.0a0 is released.
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply examples/ui_progress_updates/pyproject.toml:15]: AB#2386769

File: examples/README.md (hunk 1 above)
Comment (bkeryan) [reply examples/README.md:14]: The description is too long for the 1st line of a Python docstring as it is, so I'm removing this part

File: examples/ui_progress_updates/measurement.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+        elapsed_time_in_seconds = time.monotonic() - start_time
+
+        desired_len = int(elapsed_time_in_seconds * RANDOM_NUMBERS_PER_SECOND)
+        desired_len = max(0, min(desired_len, stop_len))
+        random_numbers.extend(_generate_random_numbers(desired_len - len(random_numbers)))
+
+        # Use yield to send incremental updates to an interactive client such as
+        # InstrumentStudio.
+        yield (elapsed_time_in_seconds, random_numbers, status)
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply examples/ui_progress_updates/measurement.py:65]: Done

**Review by dixonjoel — APPROVED** [review]

**Pull request merged**